src/server/frontend_wayland/wl_surface.cpp

```diff
@@ -66,6 +79,13 @@ mf::WlSurface::~WlSurface()
     *destroyed = true;
     if (auto session = get_session(client))
         session->destroy_buffer_stream(stream_id);
+    // it is possible to destroy a wl_surface before destroying its role (at least if the role is a subsurface)
+    // the role needs to be told its surface is invalid. Also children subsurfaces need to be unmapped.
```

Comment:
Is this comment a TODO?

If so, it should include the token "TODO", if not it isn't clear.

Reply:
This comment is not correct. Roles are destroyed when the surface is destroyed. Will edit commit to remove.

Reply:
Turns out there was a nugget of truth to that comment. I resolved the issue in 
1ae9429